executor/src/witgen/jit/symbolic_expression.rs

```diff
@@ -1,40 +1,54 @@
 use std::{
     fmt::{self, Display, Formatter},
     ops::{Add, BitAnd, Mul, Neg},
-    rc::Rc,
+    sync::Arc,
 };
 
 use num_traits::Zero;
 use powdr_number::FieldElement;
 
 use crate::witgen::range_constraints::RangeConstraint;
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum RPNExpressionElem<T: FieldElement, S> {
+    Concrete(T),
+    Symbol(S),
+    BinaryOperation(BinaryOperator),
+    UnaryOperation(UnaryOperator),
+    BitOperation(BitOperator, T::Integer),
+}
+
+/// An expression in Reverse Polish Notation.
+pub struct RPNExpression<T: FieldElement, S> {
+    pub elems: Vec<RPNExpressionElem<T, S>>,
+}
+
 /// A value that is known at run-time, defined through a complex expression
 /// involving known cells or variables and compile-time constants.
 /// Each of the sub-expressions can have its own range constraint.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq)]
```

Comment:
`PartialEq` is a recursive operation - where do you need it? Maybe it can be done based on pointers?

Reply:
Although maybe the PartialEq operation of Arc shortcuts if the pointers are the same...

Reply:
mm I don't recall why I added it, ill see if its needed

Reply:
removed